nixos/modules/services/home-automation/homebridge.nix

```diff
@@ -0,0 +1,756 @@
+      inherit (cfg.ui) host;
+    }
+    // optionalAttrs (cfg.ui.proxyHost != null) {
+      inherit (cfg.ui) proxyHost;
+    }
+    // optionalAttrs (cfg.ui.auth != null) {
+      inherit (cfg.ui) auth;
+    }
+    // optionalAttrs (cfg.ui.sessionTimeout != null) {
+      inherit (cfg.ui) sessionTimeout;
+    }
+    // optionalAttrs (cfg.ui.ssl != null) {
+      ssl = {
+        inherit (cfg.ui.ssl)
+          key
+          cert
+          pfx
+          passphrase
+          ;
+      };
+    }
+    // optionalAttrs (cfg.ui.tempUnits != null) {
+      inherit (cfg.ui) tempUnits;
+    }
+    // optionalAttrs (cfg.ui.theme != null) {
+      inherit (cfg.ui) theme;
+    }
+    // optionalAttrs (cfg.ui.lang != null) {
+      inherit (cfg.ui) lang;
+    }
+    // optionalAttrs (cfg.ui.loginWallpaper != null) {
+      inherit (cfg.ui) loginWallpaper;
+    }
+    // optionalAttrs (cfg.ui.scheduledBackupDisable != null) {
+      inherit (cfg.ui) scheduledBackupDisable;
+    }
+    // optionalAttrs (cfg.ui.scheduledBackupPath != null) {
+      inherit (cfg.ui) scheduledBackupPath;
+    }
+    // optionalAttrs (cfg.ui.debug != null) {
+      inherit (cfg.ui) debug;
+    }
+    // optionalAttrs (cfg.ui.disableServerMetricsMonitoring != null) {
+      inherit (cfg.ui) disableServerMetricsMonitoring;
+    };
+
+  defaultConfig = {
+    description = "Homebridge";
+    bridge =
+      {
+        inherit (cfg.bridge) name port;
+        username = if (cfg.bridge.username != null) then cfg.bridge.username else "CC:22:3D:E3:CE:30";
+        pin = if (cfg.bridge.pin != null) then cfg.bridge.pin else "031-45-154";
+      }
+      // optionalAttrs (cfg.bridge.advertiser != null) {
+        inherit (cfg.bridge) advertiser;
+      }
+      // optionalAttrs (cfg.bridge.bind != null) {
+        inherit (cfg.bridge) bind;
+      }
+      // optionalAttrs (cfg.bridge.setupID != null) {
+        inherit (cfg.bridge) setupID;
+      }
+      // optionalAttrs (cfg.bridge.manufacturer != null) {
+        inherit (cfg.bridge) manufacturer;
+      }
+      // optionalAttrs (cfg.bridge.model != null) {
+        inherit (cfg.bridge) model;
+      }
+      // optionalAttrs (cfg.bridge.disableIpc != null) {
+        inherit (cfg.bridge) disableIpc;
+      };
+    platforms = [
+      defaultConfigPlatform
+    ];
+  };
+
+  defaultConfigFile = pkgs.writeTextFile {
+    name = "config.json";
+    text = builtins.toJSON defaultConfig;
+  };
+in
+{
+  options.services.homebridge = with types; {
+
+    # Basic Example
+    # {
+    #   services.homebridge = {
+    #     enable = true;
+    #     # Necessary for service to be reachable
+    #     openFirewall = true;
+    #     # Most accessories need unauthenticated access
+    #     allowInsecure = true;
+    #   };
+    # }
+
+    enable = mkEnableOption "Homebridge: Homekit home automation";
+
+    bridge = mkOption {
+      description = "Homebridge Bridge options";
+      default = { };
+      type = submodule {
```

Comment:
For both this and the UI settings, I would suggest you use the `freeformType` setting here, to make your life easier when converting to JSON. It also means that this can expand without you having to manually add settings here for everything.

[Here's a recent example I did](https://github.com/NixOS/nixpkgs/blob/32ba344ba1bcf64ebfeaf742f88aa3f0c581e9fa/nixos/modules/services/torrent/cross-seed.nix#L45-L96)